src/layout/rectangles.js

```diff
@@ -0,0 +1,121 @@
+import d3 from 'd3';
+import dataJoinUtil from '../util/dataJoin';
+import {noop} from '../util/fn';
+import {range} from '../util/scale';
+import noopStrategy from './strategy/noop';
+import {rebindAll} from '../util/rebind';
+
+export default function(layoutStrategy) {
+
+    var width = d3.functor(0),
+        height = d3.functor(0),
+        x = function(d, i) { return d.x; },
+        y = function(d, i) { return d.y; };
+
+    var xScale = d3.scale.identity(),
+        yScale = d3.scale.identity(),
+        strategy = layoutStrategy || noopStrategy(),
+        component = noop;
+
+    var dataJoin = dataJoinUtil()
+        .selector('g.rectangle')
+        .element('g')
+        .attr('class', 'rectangle');
+
+    var rectangles = function(selection) {
+
+        var xRange = range(xScale),
+            yRange = range(yScale);
+        strategy.containerWidth(Math.max(xRange[0], xRange[1]))
+            .containerHeight(Math.max(yRange[0], yRange[1]));
+
+        selection.each(function(data, index) {
+            var g = dataJoin(this, data);
+
+            // obtain the rectangular bounding boxes for each child
+            var childRects = data.map(function(d, i) {
+                return {
+                    x: x(d, i),
+                    y: y(d, i),
+                    width: width(d, i),
+                    height: height(d, i)
+                };
+            });
+
+            // apply the strategy to derive the layout
+            var layout = strategy(childRects);
+
+            // offset each rectangle accordingly
+            g.attr('transform', function(d, i) {
+                var offset = layout[i];
+                return 'translate(' + offset.x + ', ' + offset.y + ')';
+            });
+
+            // set the layout width / height so that children can use SVG layout if required
+            g.attr('layout-width', function(d, i) { return childRects[i].width; });
+            g.attr('layout-height', function(d, i) { return childRects[i].height; });
+
+            g.call(component);
+        });
+    };
+
+    rebindAll(rectangles, strategy);
```

Comment:
I'm not really sure we need one - the `rectangles` component only has 4 properties, so the chance of collision is pretty small. I'm also not sure what a suitable prefix would be.